### PR TITLE
fix: Lazy Loading fixes

### LINKED ISSUE #3105: Incorrect online status of friends
When a user has more than 30 friends, and one of their second page's friends is online, after opening and closing the window, the user will be offline when actually the user is online.

## Repro steps

1. Open the friends window
2. Scroll down to load more than 1 page of friends
3. Have any friend online
4. Close the window
5. Open the window again
6. The friend that it used to be online, now its in the offline list

### LINKED ISSUE #3106: The second page of friends is loaded as soon as the window is opened
The friends window loading behaviour is not correctly working. Is triggering to load more friends (through scroll) when it shouldnt.

## Repro steps

1. Have more than 30 friends
2. Open the friends window
3. Close it
4. Open it again
5. It should only load the first 30 friends but instead is loading more

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Fixes #3105 
Fixes #3106

### Fixes included
- [x] Fixed visibility of the JumpIn button at offline friends.
- [x] Improved the visibility of the loading spinner when loading pages on friends and direct conversations.
- [x] Refactor of the search behaviour on friends and conversations.
- [x] Fixed pagination issues when scrolling down friends and conversations.
- [x] Fixed the loading of the avatar images from some conversation entries that were not being displayed correctly.
- [x] Sometimes a conversation entry was duplicated.
- [x] Prioritized the display of online friends in the list

## How to test the changes?
1. Go to https://play.decentraland.org/?renderer-branch=fix/lazy-loading-fixes
2. Check that everything related to Chat & Friends is workins correctly.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendsHUD/Scripts/FriendsHUDController.cs [hunk 1]
```diff
@@ -243,10 +249,17 @@ private void UpdateUserStatus(string userId, UserStatus status)
                 friend.blocked = IsUserBlocked(userId);
                 friends[userId] = friend;
                 View.Set(userId, friend);
+
+                if (status.presence == PresenceStatus.ONLINE)
+                    onlineFriends[userId] = friend;
+                else
+                    onlineFriends.Remove(userId);
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendsHUD/Scripts/FriendsHUDController.cs:256]: we should also consider to remove online friends on `FriendshipStatus.NOT_FRIEND` & `FriendshipStatus.REQUESTED_TO`

**Review by kwspgoliasz — CHANGES_REQUESTED** [review]
The feature appears to be implemented mostly correctly, however, several issues with the functionality of chat and friends are still present. The issues that were observed are:

1. User is not able to access a DM found with the search option - can be reproduced by searching for a DM from the channel menu: https://user-images.githubusercontent.com/110464313/193294276-ebe83de7-7364-4895-8628-81e04ff5d1a9.mp4.
2. The unfriended user is not correctly deleted from the friends list - changing the tab or closing and opening it makes the deleted friend reappear on the list. In most cases, the friend is visually on the list, however, clicking on the contextual menu gives an option to add a friend: https://user-images.githubusercontent.com/110464313/193294340-8b902013-f0b9-44a6-ad70-5d7af304f4ab.mp4.
3. The friends from other pages are loading on their own - the issue can be reproduced by opening the friends tab and waiting for about 1 minute. If the user has more than 30 friends, the issue will cause the entire friend list to load automatically: https://user-images.githubusercontent.com/110464313/193294374-30d44fa5-24f2-4616-a906-d519430e2bea.mp4.
4. Obsolete message regarding hidden chats in the chat menu - can be reproduced by going to the channel list section:

**Review by lorux0 — APPROVED** [review]

**Comment by sandrade-dcl** [comment]
@kwspgoliasz could you please re-validate again? We have already applied the fixes:

> 1. User is not able to access a DM found with the search option - can be reproduced by searching for a DM from the channel menu: https://user-images.githubusercontent.com/110464313/193294276-ebe83de7-7364-4895-8628-81e04ff5d1a9.mp4.

✔️ Fixed!

> 2. The unfriended user is not correctly deleted from the friends list - changing the tab or closing and opening it makes the deleted friend reappear on the list. In most cases, the friend is visually on the list, however, clicking on the contextual menu gives an option to add a friend: https://user-images.githubusercontent.com/110464313/193294340-8b902013-f0b9-44a6-ad70-5d7af304f4ab.mp4.

✔️ Fixed!

> 3. The friends from other pages are loading on their own - the issue can be reproduced by opening the friends tab and waiting for about 1 minute. If the user has more than 30 friends, the issue will cause the entire friend list to load automatically: https://user-images.githubusercontent.com/110464313/193294374-30d44fa5-24f2-4616-a906-d519430e2bea.mp4.

✔️ Fixed!

> 4. Obsolete message regarding hidden chats in the chat menu - can be reproduced by going to the channel list section:
>    

❓ We cannot reproduce it. Let us know if it is still happenning to you, but it wouldn't be a blocker.

**Review by kwspgoliasz — CHANGES_REQUESTED** [review]
The chat and friend features appear to be implemented mostly correctly. The basic features are functional and all of the problems mentioned in the previous comment were fixed. However, some additional issues were found during the retests:

- The DM chat history is not loaded for one user after accepting a friend request. The chat is only added after writing a new message to the friend. - https://user-images.githubusercontent.com/110464313/194051156-f8e9d7f1-eadb-468e-b071-f80856cf6c24.mp4.

- When accepting the invitation from the passport, the friend is added to the list with a noticeable delay for one user - https://user-images.githubusercontent.com/110464313/194051735-790111d0-3a19-4dd2-8c75-302f7264872a.mp4.

- In rare cases, after a friend is removed, the DM's and passport randomly do not register the update for one or both users. As a result, even though the friend was removed from the friend list, the user is described as friend in the passport. Moreover, even though the DM history is correctly deleted from the channel menu, the DM can still be accessed via the search feature. No stable repro was found for the issue, however, it seems more prone to occur, if there are at least four avatars are present in the area. After adding all of the avatars as friends and then removing them, the issue randomly occurs for one of the avatars located in the area.
#3120_Friend

**Review by kwspgoliasz — APPROVED** [review]
The ticket is approved as the most severe issues were fixed. The newly found issues are to be addressed in another ticket.

**Pull request merged**